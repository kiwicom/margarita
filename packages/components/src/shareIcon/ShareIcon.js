// @flow

import * as React from 'react';
import { Platform } from 'react-native';
import { Icon } from '@kiwicom/universal-components';

export default function ShareIcon() {
  switch (Platform.OS) {
    case 'ios': {
      return <Icon name="share-ios" />;
    }
    case 'android': {
      return <Icon name="share-android" />;
    }
    default: {
      return <Icon name="share" />;
    }
  }
}
