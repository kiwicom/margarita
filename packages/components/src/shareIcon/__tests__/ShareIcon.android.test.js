// @flow

import * as React from 'react';
import { render } from 'react-native-testing-library';

import ShareIcon from '../ShareIcon';

jest.mock('Platform', () => ({
  select: jest.fn(),
  OS: 'android',
}));

describe('ShareIcon -- android', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<ShareIcon />);
    expect(toJSON()).toMatchSnapshot();
  });
});
