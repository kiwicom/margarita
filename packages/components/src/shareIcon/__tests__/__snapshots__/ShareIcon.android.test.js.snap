// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareIcon -- android should match the snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "orbit-icons",
        "includeFontPadding": false,
        "textAlignVertical": "center",
      },
      Object {
        "fontSize": 24,
        "height": 24,
        "lineHeight": 24,
        "width": 24,
      },
      Object {
        "color": "#46515e",
      },
      undefined,
    ]
  }
>
  ÓÅ®
</Text>
`;
